File: typescript/payments/raw/retrieve.ts
import { PagouHttpClient } from "../../src/lib/http.js";
import { printResult } from "../../src/lib/format.js";
import { NotFoundError } from "../../src/lib/errors.js";
import type { Transaction } from "../../src/lib/types.js";
import { resourceIdFromArgs } from "../demo-data.js";

// Retrieves a transaction by its public UUID.
// Run: npm run pay:retrieve -- <transaction_id>
async function main(): Promise<void> {
  const id = resourceIdFromArgs("PAGOU_TRANSACTION_ID");
  const client = new PagouHttpClient();

  try {
    const { data: tx } = await client.requestData<Transaction>({
      method: "GET",
      path: `/v2/transactions/${id}`,
    });
    printResult("Transaction", {
      id: tx.id,
      status: tx.status,
      amount: tx.amount,
      paid_amount: tx.paid_amount,
      refunded_amount: tx.refunded_amount,
      paid_at: tx.paid_at,
    });
  } catch (error) {
    if (error instanceof NotFoundError) {
      console.error(`No transaction ${id}.`);
      return;
    }
    throw error;
  }
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: java/src/main/java/ai/pagou/examples/payments/Retrieve.java
package ai.pagou.examples.payments;

import ai.pagou.examples.lib.Config;
import ai.pagou.examples.lib.Errors.NotFoundException;
import ai.pagou.examples.lib.Format;
import ai.pagou.examples.lib.Models.Transaction;
import ai.pagou.examples.lib.PagouHttpClient;
import ai.pagou.examples.lib.Request;
import java.util.LinkedHashMap;
import java.util.Map;

// Retrieves a transaction by its public UUID.
// Run: mvn -q compile exec:java@pay-retrieve -Dexec.args="<transaction_id>"
public final class Retrieve {

  private Retrieve() {}

  public static void main(String[] args) {
    String id = Config.resourceIdFromArgs(args, "PAGOU_TRANSACTION_ID");
    PagouHttpClient client = new PagouHttpClient();

    try {
      Transaction tx =
          client.requestData(Request.get("/v2/transactions/" + id), Transaction.class).data();
      Map<String, Object> view = new LinkedHashMap<>();
      view.put("id", tx.id());
      view.put("status", tx.status());
      view.put("amount", tx.amount());
      view.put("paid_amount", tx.paidAmount());
      view.put("refunded_amount", tx.refundedAmount());
      view.put("paid_at", tx.paidAt());
      Format.printResult("Transaction", view);
    } catch (NotFoundException e) {
      System.err.println("No transaction " + id + ".");
    }
  }
}
